### PR TITLE
[wpt] Smarter discovery of Chrome binaries

### PR DESCRIPTION
**Pull request by Hexcles**
Previously, the default binary path for Chrome was hard-coded regardless
of the channel, making it cumbersome to test non-stable channels (in
addition to `--channel`, users had to specify `--binary`). With this
change, `./wpt run` tries to find the channel-specific binary in lieu of
the `--binary` argument:

* On Linux, find google-chrome-{stable,beta,unstable} in PATH.
* On Mac, find "Google Chrome{, Canary}.app".

(Compatible with the official Linux and macOS packages)

This change does not affect existing CI setups, because we always
specify `--binary` on CI.

Also update the doc to reflect the change.

### CODE REVIEW

File: tools/wpt/run.py [hunk 1]
```diff
@@ -351,7 +361,7 @@ def setup_kwargs(self, kwargs):
         if kwargs["binary"] is None:
             binary = self.browser.find_binary(channel=browser_channel)
             if binary:
-                kwargs["binary"] = self.browser.find_binary()
+                kwargs["binary"] = binary
```
Comment (Hexcles) [inline tools/wpt/run.py]: Note: this is an unrelated drive-by fix.

**Review by sideshowbarker — APPROVED** [review]
Reviewed code changes and doc changes, tested (on macOS)

**Comment by Hexcles** [comment]
Thanks for the review!
